docs/node-mixin/dashboards/node.libsonnet

```diff
@@ -16,25 +16,30 @@ local gauge = promgrafonnet.gauge;
           datasource='$datasource',
           span=6,
           format='percentunit',
-          max=100,
+          max=1,
           min=0,
+          stack=true,
         )
         .addTarget(prometheus.target(
-          // TODO: Consider using `${__interval}` as range and a 1m min step.
           |||
-            1 - rate(node_cpu_seconds_total{%(nodeExporterSelector)s, mode="idle", instance="$instance"}[1m])
+            (
+              (1 - rate(node_cpu_seconds_total{%(nodeExporterSelector)s, mode="idle", instance="$instance"}[$__interval]))
+            / ignoring(cpu) group_left
```

Comment:
@beorn7 I just came across this while trying to figure out why my fully utilized system only shows 12.5% cpu usage for each cpu. I think this doesn't make sense, right? See also slack: https://cloud-native.slack.com/archives/C01AUBA4PFE/p1725300066105619

Reply:
I have hardly any memories of this.

I can only speculate that this past version of myself either observed or merely assumed that each individual CPU would count seconds just for itself. So a fully utilized 8 core system should see 8 CPU seconds counted up in total per second.

Therefore, this query divides by the number of CPUs, so that the resulting value will be 1 for a system that fully utilizes all its CPUs.

Thinking about it, I think this should be the case. What system are you running your query on? Is it maybe some virtual machine that pretends to have 8 cores but is only backed by 1 physical core?

Reply:
I could also see hyperthreading being an issue here.